EssentialFeed/EssentialFeedTests/Image Comments API /LoadImageCommentsFromRemoteUseCaseTests.swift

```diff
@@ -0,0 +1,193 @@
+//
+//  RemoteImageCommentLoaderTests.swift
+//  EssentialFeedTests
+//
+//  Created by Ángel Vázquez on 17/03/21.
+//  Copyright © 2021 Essential Developer. All rights reserved.
+//
+
+import EssentialFeed
+import XCTest
+
+class LoadImageCommentsFromRemoteUseCaseTests: XCTestCase {
+	func test_init_doesNotRequestDataFromURL() {
+		let (_, client) = makeSUT()
+		
+		XCTAssertTrue(client.requestedURLs.isEmpty)
+	}
+	
+	func test_load_requestsDataFromURL() {
+		let (sut, client) = makeSUT()
+		let url = anyURL()
+		
+		_ = sut.load(from: url) { _ in }
+		
+		XCTAssertEqual(client.requestedURLs, [url])
+	}
+	
+	func test_loadTwice_requestDataFromURLTwice() {
+		let (sut, client) = makeSUT()
+		let url = anyURL()
+		
+		_ = sut.load(from: url) { _ in }
+		_ = sut.load(from: url) { _ in }
+		
+		XCTAssertEqual(client.requestedURLs, [url, url])
+	}
+	
+	func test_load_deliversConnectivityErrorOnClientError() {
+		let (sut, client) = makeSUT()
+		let error = anyNSError()
+		
+		expect(sut, toCompleteWith: failure(.connectivity), when: {
+			client.complete(with: error)
+		})
+	}
+	
+	func test_load_deliversInvalidDataErrorOnNon2xxHTTPResponse() {
+		let (sut, client) = makeSUT()
+		let samples = [199, 300, 400, 500]
+		
+		samples.enumerated().forEach { index, code in
+			expect(sut, toCompleteWith: failure(.invalidData), when: {
+				client.complete(withStatusCode: code, data: anyData(), at: index)
```

Comment:
`anyData` is not a valid response for this endpoint based on API the contract. It should use a valid response to be able to test the `status code` logic.

It could be an empty JSON list: 

```suggestion
				client.complete(withStatusCode: code, data: makeCommentsJSON([]), at: index)
```